src/Open3D/IO/FileFormat/FilePTS.cpp

```diff
@@ -118,7 +118,8 @@ bool WritePointCloudToPTS(const std::string &filename,
             fprintf(file, "%.10f %.10f %.10f\r\n", point(0), point(1),
                     point(2));
         } else {
-            const auto &color = pointcloud.colors_[i] * 255.0;
+            const auto &color = pointcloud.colors_[i] * 255.0 +
+                                Eigen::Vector3d(0.5, 0.5, 0.5);
```

Comment:
could you remind me why we need to add 0.5?

Reply:
floating point error, various ops will produce slightly smaller double (for example, saving as .xyzrgb, then loading that) which when converted back to uint8_t will end up a different number.  I added UnitTests for various tests like this, and then decided to explicitly check for rounding as well.  Turns out we had this fp error in other formats, so I fixed all of those as well.